Scuffed DeepWalk beskrivelse - Synes den er svær at gå i detaljer med (#49)
thesis20/report: sections/graphrecs/deepwalk.tex

\subsection{DeepWalk}

> Deepwalk is an algorithm that uses a deep learning technique to learn social representations of a graph's vertices\cite{DeepWalk}.

Reviewer: ```suggestion
DeepWalk is an algorithm that uses a deep learning technique to learn social representations of a graph's vertices\cite{DeepWalk}.
```

> Deepwalk is an algorithm that uses a deep learning technique to learn social representations of a graph's vertices\cite{DeepWalk}.

Reviewer 2: ```suggestion
DeepWalk is an algorithm that uses a deep learning technique to learn social representations of a graph's nodes\cite{DeepWalk}.
```

> It makes use of neural language models where the language consists of "sentences" generated by doing random walks between the vertices of the graph.

Reviewer: er det neural eller natural????

Author: det er neural!!!!!

> It makes use of neural language models where the language consists of "sentences" generated by doing random walks between the vertices of the graph.

Reviewer 2: ```suggestion
It makes use of neural language models where the language consists of "sentences" generated by doing random walks between the nodes of the graph.
```

> By doing this, latent features that capture the neighborhood similarity of the vertices can be learned.

Reviewer 2: ```suggestion
By doing this, latent features that capture the neighborhood similarity of the nodes can be learned.
```

> This results in a low dimensional vector representation of each vertex in the graph.

Reviewer 2: ```suggestion
This results in a low dimensional vector representation of each node in the graph.
```

> The random walks are done a number of times for each vertex $v_i$.

Reviewer 2: ```suggestion
The random walks are done a number of times for each node $v_i$.
```

> The nodes in the walk are chosen at random to form the random walks $W_{vi}$ which are used as "sentences" in the neural language model and the vertices are the "words" of the sentences.

Reviewer 2: ```suggestion
The nodes in the walk are chosen at random to form the random walks $W_{vi}$ which are used as "sentences" in the neural language model and the nodes are the "words" of the sentences.
```

> The vertices in the walk are chosen at random to form the random walks $W_{vi}$ which are used as "sentences" in the neural language model and the vertices are the "words" of the sentences.

Reviewer 2: ```suggestion
The nodes in the walk are chosen at random to form the random walks $W_{vi}$ which are used as "sentences" in the neural language model and the vertices are the "words" of the sentences.
```

> The language model is then used to estimate the likelihood that vertices appear in the same random walk.

Reviewer 2: ```suggestion
The language model is then used to estimate the likelihood that nodes appear in the same random walk.
```

> This is done using the language model SkipGram, and the result of this is a vector representation of each vertex in d-dimensional space.

Reviewer: ```suggestion
This is done using the language model SkipGram, and the result of this is a vector representation of each vertex in $d$-dimensional space.
```

> This is done using the language model SkipGram, and the result of this is a vector representation of each vertex in $d$-dimensional space.

Reviewer 2: ```suggestion
This is done using the language model SkipGram, and the result of this is a vector representation of each node in $d$-dimensional space.
```

\\\\
The next step is to then use the vector representations to generate recommendations.

> Here we opted to go for a k-nearest neighbor (kNN) approach where we find the 10 nearest user-vertices to a user vertex $u$ using the vector representations.

Reviewer 2: ```suggestion
Here we opted to go for a k-nearest neighbor (kNN) approach where we find the 10 nearest user nodes to a user node $u$ using the vector representations.
```

> Here we opted to go for a k-nearest neighbor (kNN) approach where we find the 10 nearest neighbor user-vertices to a user vertex $u$ using the vector representations.

Reviewer: ```suggestion
Here we opted to go for a k-nearest neighbor (kNN) approach where we find the 10 nearest user-vertices to a user vertex $u$ using the vector representations.
```

> To then generate a rating prediction for a specific item $i$, we look at whether the 10 neighbors have rated item $i$ and use the average of their ratings of $i$ as the predicted rating for user $u$ to item $i$.

Reviewer: ```suggestion
To then generate a rating prediction for a specific item $i$, we look at whether the 10 neighbors have rated item $i$ and use the average of their ratings of $i$ as the predicted rating for user $u$ on item $i$.
```
